### DISCUSSION TITLE
Warnings upon Compilation

### DISCUSSION THREAD

**Discussion by VickyMerzOwn** [carbon-language/carbon-lang#1987] [General]
Getting these warnings on compiling any *.carbon file using the `bazel run //explorer -- ./test.carbon` command. 

```
WARNING: Download from https://mirror.bazel.build/ftp.gnu.org/gnu/m4/m4-1.4.18.tar.xz failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://mirror.bazel.build/github.com/jmillikin/rules_m4/releases/download/v0.1/m4-gnulib-788db09a9f88abbef73c97e8d7291c40455336d8.tar.xz failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://mirror.bazel.build/ftp.gnu.org/gnu/bison/bison-3.3.2.tar.xz failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://mirror.bazel.build/github.com/jmillikin/rules_bison/releases/download/v0.1/bison-gnulib-788db09a9f88abbef73c97e8d7291c40455336d8.tar.xz failed: class java.io.FileNotFoundException GET returned 404 Not Found
```

The warning isn't affecting anything, as of now at least.

Our professor (@c-i-p-h-e-r) is making us use Carbon for programming assignments in his course to ensure a level playground for non CS majors taking his course (CS553 - Cryptography, IIT Bhilai, India). Don't see many discussions on StackOverflow or any other forums. So here I am browsing through your discussions and issues. An entire class would be grateful on getting a swift response.

**Comment by jonmeow** [comment]
These warnings are coming from Bazel, I don't think there's much to do about it. They're downloaded successfully directly.

> Our professor (@c-i-p-h-e-r) is making us use Carbon for programming assignments in his course to ensure a level playground for non CS majors taking his course (CS553 - Cryptography, IIT Bhilai, India). 

Please ask them not to do this. Carbon's really not ready for use. It'd be better to use an established programming language. Given crypto, I'd encourage something like C/C++, Rust, Swift, Zig, etc as a good choice for something like this. Carbon's really too early to be used for a course (https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/faq.md#how-soon-can-we-use-carbon)

**Reply by VickyMerzOwn** [reply]
> These warnings are coming from Bazel, I don't think there's much to do about it. They're downloaded successfully directly.

Oh okay, thanks though!

> Please ask them not to do this. Carbon's really not ready for use. It'd be better to use an established programming language. Given crypto, I'd encourage something like C/C++, Rust, Swift, Zig, etc as a good choice for something like this. Carbon's really too early to be used for a course (https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/faq.md#how-soon-can-we-use-carbon)

Will share this discussion with him. He must've considered a few things on his own. Let's see what happens...

Thank You for the quick reply!
